node-tests/e2e/fastboot-support-test.js

```diff
@@ -110,6 +110,30 @@ describe('e2e: fastboot integration', function () {
     });
   });
 
+  describe('scenario: with reportOnly = false', function () {
+    before(async function () {
+      await setConfig(testProject, { reportOnly: false });
+      await testProject.runEmberCommand('build');
```

Comment:
> I added a test. It doesn't actually fail with the previous code, since assert only fails in development mode, but as I said I have no clue how to make the test server behave like a dev environment would.

You can pass flags like build environment to the command:

```suggestion
      await testProject.runEmberCommand('build', '--environment', 'production');
```

But `development` should be the default environment: https://cli.emberjs.com/release/advanced-use/cli-commands-reference/#emberbuild

Reply:
Please see my new suggestion about asserting the HTTP status code.